engine/storage/image/src/main/java/org/apache/cloudstack/storage/image/TemplateDataFactoryImpl.java

```diff
@@ -204,11 +205,9 @@ public List<TemplateInfo> listTemplateOnCache(long templateId) {
      */
     private Long getOneMatchingPoolIdFromRefs(List<VMTemplateStoragePoolVO> existingRefs, List<StoragePoolVO> pools) {
         if (pools.isEmpty()) {
-            throw new CloudRuntimeException("No storage pools found");
+            return null;
```

Comment:
returning null is not a good practice. Can this `if (pools.isEmpty())` be handled in the calling methods and an exception be thrown here?

Reply:
Refactored